zerver/tests/test_digest.py

```diff
@@ -71,6 +62,89 @@ def test_multiple_stream_senders(self,
         self.assertIn('some content', teaser_messages[0]['content'][0]['plain'])
         self.assertIn(teaser_messages[0]['sender'], expected_participants)
 
+    @mock.patch('zerver.lib.digest.enough_traffic')
+    @mock.patch('zerver.lib.digest.send_future_email')
+    def test_soft_deactivated_user_multiple_stream_senders(self,
+                                                           mock_send_future_email: mock.MagicMock,
+                                                           mock_enough_traffic: mock.MagicMock) -> None:
+
+        one_day_ago = timezone_now() - datetime.timedelta(days=1)
+        Message.objects.all().update(pub_date=one_day_ago)
+
+        othello = self.example_user('othello')
+        for stream in ['Verona', 'Scotland', 'Denmark']:
+            self.subscribe(othello, stream)
+        RealmAuditLog.objects.all().delete()
+
+        othello.long_term_idle = True
+        othello.save(update_fields=['long_term_idle'])
+
+        # Send messages to a stream and unsubscribe - subscribe from that stream
+        senders = ['hamlet', 'cordelia',  'iago', 'prospero', 'ZOE']
+        self.simulate_stream_conversation('Verona', senders)
+        self.unsubscribe(othello, 'Verona')
+        self.subscribe(othello, 'Verona')
+
+        # Send messages to other streams
+        self.simulate_stream_conversation('Scotland', senders)
+        self.simulate_stream_conversation('Denmark', senders)
+
+        one_sec_ago = timezone_now() - datetime.timedelta(seconds=1)
```

Comment:
We should do an hour for this -- otherwise, the test will flake if the steps above took longer than a second, which they could on a slow/hosed machine.